### ISSUE TITLE
Failed to create checkpoint with "Parquet does not support writing empty structs"

### ISSUE THREAD

**Issue by rtyler** [delta-io/delta-rs#2189]
# Environment

**Delta-rs version**: 0.16.5

**Binding**:  rust

**Environment**: Linux/amd64
- **Cloud provider**: AWS
- **OS**:
- **Other**:

***
# Bug

**What happened**:

Spotted in some production logs, I need to figure out how this is possible: 

```
ERROR Failed to create checkpoint for s3://REDACTED ParquetParseError { source: ArrowError("Parquet does not support writing empty structs") }    
```

**What you expected to happen**:

A checkpoint to be written :smile: 

**How to reproduce it**:

**More details**:

**Comment by rtyler** [comment]
I believe this is coming out of the `parquet_bytes_from_state` function, which doesn't have any tests :scream:

**Comment by rtyler** [comment]
What is unlucky for me is that I took the transaction log from a table in production, 0.17 is able to checkpoint it, but 0.16.5 is not. Which means its identical to what is fixed by #2125

Unfortunately, that likely means I will have to incur the migration cost of the new DynamoDB changes for this application :laughing:
